pkg/storage/etcd3/watcher.go

```diff
@@ -182,6 +194,10 @@ func (wc *watchChan) startWatching() {
 			wc.sendEvent(parseEvent(e))
 		}
 	}
+	// When we come to this point, it's possible that:
+	// - client side ends the watch, e.g. cancel the context, close the client.
+	// - server side ends the watch.
+	close(watchClosedCh)
```

Comment:
It shouldn't always happen --  it should happen only if watch chan is closed and no error message returned from watch chan.